rudder-server-root/SOURCES/rudder-server-root.init

```diff
@@ -32,67 +32,188 @@
 
 # Variables
 
-## Init scripts
-APACHE2_INIT=/etc/init.d/apache2
-if [ ! -x ${APACHE2_INIT} ] && [ -x /etc/init.d/httpd ]; then
-	APACHE2_INIT=/etc/init.d/httpd
+## Colors configuration
+ECHO_BIN="/bin/echo -e"
+
+### Enable colors only if stdout is a terminal
+if [ -t 1 ]; then
+  GREEN="\\033[1;32m"
+  RED="\\033[1;31m"
+  BLUE="\\033[1;34m"
+  PINK="\\033[1;35m"
+  WHITE="\\033[0;02m"
+  WHITELIGHT="\\033[1;08m"
+  YELLOW="\\033[1;33m"
+  CYAN="\\033[1;36m"
+  NORMAL="\\033[0;39m"
+  COPYPASTE="${ECHO_BIN} ${GREEN}---8<---${NORMAL}"
+else
+  GREEN=""
+  RED=""
+  BLUE=""
+  PINK=""
+  WHITE=""
+  WHITELIGHT=""
+  YELLOW=""
+  CYAN=""
+  NORMAL=""
+  COPYPASTE="${ECHO_BIN} ---8<---"
 fi
-SLAPD_INIT=/etc/init.d/rudder-slapd
-JETTY_INIT=/etc/init.d/rudder-jetty
-RUDDER_AGENT_INIT=/etc/init.d/rudder-agent
-POSTGRESQL_INIT=/etc/init.d/postgresql
+
+## Active roles
+
+HAVE_SLAPD=0
+HAVE_POSTGRESQL=0
+HAVE_JETTY=0
+HAVE_AGENT=1
+
+### Check slapd's status if we are rudder-inventory-ldap
+[ -e /opt/rudder/etc/server-roles.d/rudder-inventory-ldap ] && HAVE_SLAPD=1
+
+### Check PostgreSQL's status if we are rudder-db
+[ -e /opt/rudder/etc/server-roles.d/rudder-db ] && HAVE_POSTGRESQL=1
+
+### Check Jetty's status if we are rudder-webapp or rudder-inventory-endpoint
+[ -e /opt/rudder/etc/server-roles.d/rudder-inventory-endpoint ] && HAVE_JETTY=1
+[ -e /opt/rudder/etc/server-roles.d/rudder-webapp ] && HAVE_JETTY=1
+
+## Service names
+APACHE2=apache2
+if [ ! -x /etc/debian_version ] && [ ! -x /etc/SuSE-release ]; then
```

Comment:
Not really. We used to check for the init.d script, but it would not work with non-sysvinit systems...

Do you have a suggestion for this ?